src/components/header/search.jsx

```diff
@@ -107,26 +104,37 @@ export default function Search() {
 			// The <loading-bar> is sat alongside the router & has it's state controlled by it,
 			// so while we could create a new context to be able to set it here, direct DOM
 			// manipulation is a heck of a lot simpler.
-			loadingBar.setAttribute('showing', 'true');
+			document.querySelector('loading-bar')?.setAttribute('showing', 'true');
 		}
 	};
 
 	return (
 		<div class={style.search} ref={root}>
-			{/* Copy/paste of the HTML DocSearch normally generates, used as a placeholder */}
-			<button
-				type="button"
-				aria-label="Search"
-				class="DocSearch DocSearch-Button"
-				onMouseOver={loadDocSearch}
-				onTouchStart={loadDocSearch}
-				onFocus={loadDocSearch}
-				onClick={onInteraction}
-			>
-				<span class="DocSearch-Button-Container">
-					<span class="DocSearch-Button-Placeholder">Search</span>
-				</span>
-			</button>
+			{rendered.current ? (
+				<ErrorBoundary>
+					<DocSearch
+						apiKey={config.docsearch.apiKey}
+						indexName={config.docsearch.indexName}
+						appId={config.docsearch.appId}
+						transformItems={transformItems}
+					/>
+				</ErrorBoundary>
+			) : (
```

Comment:
After a user interacts with the search, such as clicking a link, it'll no longer function properly and the modal will not pop up as navigating rerenders the header and returns the placeholder content, wiping out the newly-rendered `<DocSearch>`.